src/platforms/dialogflow/google/apis/DigitalGoods.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  GoogleActionsTransactionsV3CompletePurchaseValuePurchaseStatus,
+  GoogleActionsTransactionsV3SkuIdSkuType,
+  GoogleCloudDialogflowV2WebhookRequest,
+} from "actions-on-google";
+import { LambdaLog } from "lambda-log";
+import * as _ from "lodash";
+import * as rp from "request-promise";
+
+import { ApiBase } from "./ApiBase";
+import { ITransactionOptions } from "./ITransactionOptions";
+
+/**
+ * DigitalGoods API class reference
+ * https://developers.google.com/actions/transactions/digital/dev-guide-digital
+ */
+export class DigitalGoods extends ApiBase {
+  public tag: string = "DigitalGoods";
+
+  constructor(
+    event: GoogleCloudDialogflowV2WebhookRequest,
+    log: LambdaLog,
+    options: ITransactionOptions,
+  ) {
+    super(event, log, options);
+
+    this.tag = "DigitalGoods";
+  }
+
+  /**
+   * Gets Entitlements from PlayStore
+   */
+  public async getInAppEntitlements(skus: string[]) {
+    return this.getSkus(skus, "SKU_TYPE_IN_APP");
+  }
+
+  /**
+   * Gets Subscriptions from PlayStore
+   */
+  public async getSubscriptions(skus: string[]) {
+    return this.getSkus(skus, "SKU_TYPE_SUBSCRIPTION");
+  }
+
+  /**
+   * Gets Skus from PlayStore
+   */
+  public async getSkus(skus: string[], type: GoogleActionsTransactionsV3SkuIdSkuType) {
+    const conversationId = _.get(this.rawEvent, "originalDetectIntentRequest.payload.conversation.conversationId");
+    const androidAppPackageName = _.get(this.transactionOptions, "androidAppPackageName");
+
+    if (!androidAppPackageName) {
+      throw new Error("Android App package name missing");
+    }
+
+    const credentials = await this.getCredentials();
+    const bearer = credentials.access_token as string;
+
+    const options = {
+      auth: {
+        bearer,
+      },
+      body: {
+        conversationId,
+        ids: skus,
+        skuType: type,
+      },
+      json: true, // Automatically parses the JSON string in the response
+      method: "POST",
+      uri: `https://actions.googleapis.com/v3/packages/${androidAppPackageName}/skus:batchGet`,
+    };
+
+    return rp(options);
+  }
+
+  /**
+   * Gets Google Transactions Status
+   */
+  public getPurchaseStatus(): GoogleActionsTransactionsV3CompletePurchaseValuePurchaseStatus {
+    const googleArguments = _.get(this.rawEvent, "originalDetectIntentRequest.payload.inputs[0].arguments", []);
+    let purchaseStatus: GoogleActionsTransactionsV3CompletePurchaseValuePurchaseStatus = "PURCHASE_STATUS_UNSPECIFIED";
+
+    _.forEach(googleArguments, (argument) => {
+      if (argument.name === "COMPLETE_PURCHASE_VALUE") {
+        purchaseStatus = argument.extension.purchaseStatus;
+
+        return false;
+      }
+    });
+
+    return purchaseStatus;
+  }
+
+  /**
+   * Checks if purchase status is PURCHASE_STATUS_ALREADY_OWNED
+   */
+  public isPurchaseStatusAlreadyOwned(): boolean {
+    const purchaseStatus = this.getPurchaseStatus();
+
+    return purchaseStatus === "PURCHASE_STATUS_ALREADY_OWNED";
```

Comment:
Do you think you can create an object with all this types?

"PURCHASE_STATUS_ALREADY_OWNED"
"PURCHASE_STATUS_ITEM_CHANGE_REQUESTED"


DIGITAL_GOOD_STATUS = ... (you can use an enum)